### PR TITLE
Fellowships directory pages

### PR DESCRIPTION
**Pull request by mmmavis**
Related to #955

- latest Sketch file: https://drive.google.com/drive/folders/13B0EMMavSNsibacHIpJd1sMi9TvslR7o (v2 one)
- feel free to create some fake profiles on Pulse API staging for testing: https://pulse-api.mofostaging.net/admin/profiles/userprofile/
- [to be fixed] API currently doesn't return `location` and `profile id` info. That's why the location field and "see work" link in the `<Person>` cards don't work as expected.

e.g., 
screen shot 2018-02-09 at 11 51 45 am

---

🔷  https://foundation-mofostaging-pr-1046.herokuapp.com/fellowships/directory/
- this page shows all 2017 fellows

---

🔷 https://foundation-mofostaging-pr-1046.herokuapp.com/fellowships/directory/<program_type>
- shows all fellows on that program
- fellows are grouped by program year
- e.g., https://foundation-mofostaging-pr-1046.herokuapp.com/fellowships/directory/senior

### CODE REVIEW

**Comment by mmmavis** [comment]
@alanmoo assigned this one to your for review since you were already checking out this directory page work earlier

File: source/js/main.js [hunk 1]
```diff
@@ -23,13 +23,14 @@ import fellowships from './fellowships';
 const SHOW_MEMBER_NOTICE = false;
 
 // To be populated via XHR...
-let env, networkSiteURL;
+let env, networkSiteURL, pulseApiURL;
```
Comment (alanmoo) [inline source/js/main.js]: Is this going to cause problems when promoting heroku pipelines?

Comment (alanmoo) [reply source/js/main.js]: Oh or is it pulling this from the environment json view that the server exposes?

Comment (mmmavis) [reply source/js/main.js]: @alanmoo I think it should be fine. I'm following exactly how we are doing to get `networkSiteURL` from the env var `NETWORK_SITE_URL`.

File: network-api/networkapi/fellows/urls.py [hunk 2]
```diff
@@ -3,11 +3,26 @@
 
 urlpatterns = [
     url(r'^$', views.fellows_home, name='fellowships-home'),
+    url(r'^science/$', views.fellows_science, name='fellowships-science'),
+    url(r'^open-web/$', views.fellows_openweb, name='fellowships-open-web'),
     url(r'^directory/$',
         views.fellows_directory,
         name='fellowships-directory'),
+    url(r'^directory/senior$',
```
Comment (alanmoo) [inline network-api/networkapi/fellows/urls.py]: I know we're doing things pretty statically, but I think it's worth it to make this DRYer. Check out [named groups](https://djangobook.com/advanced-views-urlconfs/#leanpub-auto-named-groups)

File: source/js/fellowships.js [hunk 3]
```diff
@@ -5,8 +5,146 @@ import ReactDOM from 'react-dom';
...
+
+  req.addEventListener(`load`, () => {
+    callback.call(this, JSON.parse(req.response));
+  });
+
+  req.open(`GET`, `https://${pulseApiDomain}/api/pulse/profiles/?${queryString}`);
+  req.send();
+}
+
+function renderFellowCard(fellow) {
```
Comment (alanmoo) [inline source/js/fellowships.js]: Can you put a comment in describing what this is doing (vs, say, the existing `person.jsx` component)

**Comment by mmmavis** [comment]
💃

File: source/js/fellowships.js [hunk 4]
```diff
@@ -5,8 +5,146 @@ import ReactDOM from 'react-dom';
...
+      onClick={(event) => { window.scrollTo(0, document.getElementById(event.target.dataset.targetId).offsetTop); }}
+      data-target-id={`fellowships-directory-${getTypeSlug(option)}`}
+    >
+      {`${option}${option === `senior` ? ` Fellow` :``}`}
+    </button>;
+  };
+
+  getFellows({'program_year': `2017`}, fellows => {
+    let fellowsByType = groupFellowsByAttr(`program_type`, fellows);
+    const ORDER = [ `senior`, `science`, `open web`, `tech policy`, `media`];
```
Comment (alanmoo) [inline source/js/fellowships.js]: Maybe bring this to the top while you're at it?

File: network-api/networkapi/fellows/urls.py [hunk 5]
```diff
@@ -3,11 +3,14 @@
...
 urlpatterns = [
     url(r'^$', views.fellows_home, name='fellowships-home'),
+    url(r'^science/$', views.fellows_science, name='fellowships-science'),
+    url(r'^open-web/$', views.fellows_openweb, name='fellowships-open-web'),
     url(r'^directory/$',
         views.fellows_directory,
         name='fellowships-directory'),
+    url(r'^directory/(?P<program_type_slug>[-\w]+)/$',
+        views.fellows_directoy_type,
+        name='fellowships-directory-senior'),
```
Comment (alanmoo) [inline network-api/networkapi/fellows/urls.py]: why is this called "senior"?

File: network-api/networkapi/templates/fellowships-base.html [hunk 6]
```diff
@@ -80,7 +80,7 @@
           <div class="col">
             <div id="multipage-nav" class="d-flex flex-row align-items-center">
               <div><a href="{% url 'fellowships-home' %}" class="{% fellowship_active_nav request 'fellowships-home fellowships-science fellowships-open-web' %}">Fellowships</a></div>
-              <div><a href="{% url 'fellowships-directory' %}" class="{% fellowship_active_nav request 'fellowships-directory' %}">Directory</a></div>
+              <div><a href="{% url 'fellowships-directory' %}" class="{% fellowship_active_nav request 'fellowships-directory fellowships-directory-senior' %}">Directory</a></div>
```
Comment (alanmoo) [inline network-api/networkapi/templates/fellowships-base.html]: Similar to above, I'm not clear on what the `senior` bit is doing here

**Pull request merged**